recipes/data-prep-kit/end-2-end-code-data-prep-finetuning.ipynb

```diff
@@ -0,0 +1,869 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a href=\"https://colab.research.google.com/github/IBM/data-prep-kit/blob/dev/examples/notebooks/code/sample-notebook.ipynb\">\n",
+    "  <img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/>\n",
+    "</a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "_-NOkuTxiP7r",
+    "outputId": "043f32fc-c476-433e-86b6-d7e9abd4d285"
+   },
+   "source": [
+    "This demo notebook shows how to use [data-prep-kit](https://github.com/IBM/data-prep-kit) to build a data preparation pipeline that can be used for fine tuning or extended pre-training. We will discuss the various data preparation steps to process raw data (code repositories), tokenise it that can then be fine tuned using any popular code models. We will also discuss a novel recipe for semantic ordering of files in a repository which has shown to enhance model training. Please see our [paper](https://arxiv.org/abs/2407.13739) here for more details. For this demo, we will use the [codeparrot/github-code](https://huggingface.co/datasets/codeparrot/github-code) dataset hosted on Hugging Face datasets. \n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "Install data-prep-toolkit and datasets library. This notebook requires atleast 8 cpus. \n",
+    "To run on google colab, it is recommended to change the runtime to TPUs to get the required number of cpus.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%capture logpip --no-stderr\n",
+    "!pip install data-prep-toolkit-transforms-ray==0.2.1.dev1\n",
+    "!pip install datasets\n",
+    "!pip install pandas"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "8VhIsZViaU2i"
+   },
+   "source": [
+    "We use parallel processing capability using Ray, so that beyond the demo, a user can also use this for actual production runs on larger datasets, with minor code changes. Please read [here](https://github.com/IBM/data-prep-kit?tab=readme-ov-file#-about-) on various features of data-prep-kit that includes flexibility of compute to run from laptop to cluster.  There are three parameters, that the user can change, as per usecase:\n",
+    "\n",
+    "`runtime_num_worker`: number of parallel workers to be used\n",
+    "\n",
+    "`num_cpus`: number of cpus to be used per worker\n",
+    "\n",
+    "`run_locally: True` start a ray cluster for parallel computation\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "J_UbnF9wbj95"
+   },
+   "outputs": [],
+   "source": [
+    "from data_processing_ray.runtime.ray import RayTransformLauncher\n",
```

Comment:
this line fails in testing and also fails for me when trying to run in colab

Reply:
I have been able to run it in colab post the pip installations. Please see link here https://colab.research.google.com/drive/1uxKhvPZLbSnPmAGYfQ0jnrJIiDbfeIcJ#scrollTo=NbF_Zw3KBazf

Reply:
i don't have access, but was your runtime CPU only or TPU?